scipy/optimize/_linprog.py

```diff
@@ -1072,12 +1345,77 @@ def linprog(c, A_ub=None, b_ub=None, A_eq=None, b_eq=None,
     meth = method.lower()
     if options is None:
         options = {}
-
-    if meth == 'simplex':
-        return _linprog_simplex(c, A_ub=A_ub, b_ub=b_ub, A_eq=A_eq, b_eq=b_eq,
-                                bounds=bounds, callback=callback, **options)
-    elif meth == 'interior-point':
-        return _linprog_ip(c, A_ub=A_ub, b_ub=b_ub, A_eq=A_eq, b_eq=b_eq,
-                           bounds=bounds, callback=callback, **options)
-    else:
-        raise ValueError('Unknown solver %s' % method)
+    
+    # This needs to be before presolve for the tests to pass.
+    # If the problem is solved in presolve the error will not be raised
+    # leading the test to fail. 
+    # Should the check be moved to raise a ``NotImplementedError``
+    # only if callback would actually have been used?
+    if meth == 'interior-point' and callback:
+        raise NotImplementedError("method 'interior-point' does not support "
+                                  "callback functions.")
+
+    default_tol = 1e-12 if meth == 'simplex' else 1e-9
+    tol = options.get('tol', default_tol)
+
+    # This is an undocumented option for unit testing sparse presolve
+    _sparse_presolve = options.pop('_sparse_presolve', False)
```

Comment:
`_sparse_presolve` applies only to this part of the program (changing A to sparse for testing). Previously this was done in `_linprog_ip(...)` with `_sparse_presolve` passed as a (undocumented) option. The conversion now occurs in  `_linprog(...)`. `_linprog_ip(...)` no longer requires `_sparse_presolve` to be passed as an option, and has been removed.  If `options` still contains `_sparse_presolve`  it will be passed to `_check_unknown_options(...)`, not be recognized as an option (as it no longer is) and raise an error.